imperial_coldfront_plugin/gpfs_client.py

```diff
@@ -219,16 +258,33 @@ def _create_fileset_set_quota(
     fileset_name: str,
     owner_id: str,
     group_id: str,
-    path: str,
+    path: Path,
     permissions: str,
     block_quota: str,
     files_quota: str,
+    parent_fileset: str,
 ):
     """Create a fileset and set a quota in the requested filesystem."""
     client = GPFSClient()
+    try:
+        client.create_fileset_directory(
+            filesystem_name,
+            parent_fileset,
+            str(path.parent.relative_to(path.parents[2])),
+        )
+    except ErrorWhenProcessingJob:
+        # will fail if directory already exists
+        pass
```

Comment:
Do we want to silently ignore these errors? There might be other reasons why this fails, I guess... At the very least, I would log the error somewhere.

Reply:
Good shout. I've added a check on the return value to check for the reason. Have also moved the logic up into the client.